src/main/java/io/harness/cf/client/api/InnerClient.java

```diff
@@ -241,7 +241,7 @@ public void update(@NonNull final Message message, final boolean manual) {
   }
 
   private synchronized void initialize(@NonNull final Processor processor) {
-    if (closing) {
+    if (initialized || closing) {
```

Comment:
I think it maybe worth inserting a debug, or possible warn log message to indicate we're already initialised or closing.
```java
log.debug("client is already intitialized {} or closing {}", initialized, closing)
```